### PR TITLE
Update tracking script.

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #8944 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Core/Conversion_Tracking/Conversion_Tracking.php [hunk 1]
```diff
@@ -131,8 +131,25 @@ function ( Conversion_Events_Provider $active_provider ) {
...
+
+				if ( !!window._googlesitekit.throttledEvents[key] ) {
+					return;
+				}
+				window._googlesitekit.throttledEvents[key] = true;
+				setTimeout( () => {
+					delete window._googlesitekit.throttledEvents[key];
+				}, 500 );
+
+				gtag("event", name, {...data, event_source: "site-kit" });
```
Comment (tofumatt) [inline includes/Core/Conversion_Tracking/Conversion_Tracking.php:148]: Super nitpick, but:

```suggestion
				gtag( "event", name, { ...data, event_source: "site-kit" } );
```

😅

File: includes/Core/Conversion_Tracking/Conversion_Tracking.php [hunk 2]
```diff
@@ -131,8 +131,25 @@ function ( Conversion_Events_Provider $active_provider ) {
...
+			window._googlesitekit.gtagEvent = (name, data) => {
+				var key = JSON.stringify( { name, data } );
+
+				if ( !!window._googlesitekit.throttledEvents[key] ) {
+					return;
+				}
+				window._googlesitekit.throttledEvents[key] = true;
+				setTimeout( () => {
+					delete window._googlesitekit.throttledEvents[key];
+				}, 500 );
```
Comment (tofumatt) [inline includes/Core/Conversion_Tracking/Conversion_Tracking.php:146]: Why such a long timeout here? If a plugin tracks an event multiple times in a short period of time, it seems we'll end up ignoring them. Half a second is brief, but it's totally possible to, for instance, add multiple items to cart in that time by repeatedly clicking on an "Add to cart" button.

Is there any reason we couldn't make this much briefer? Say 10/20/50ms? 🤷🏻‍♂️

**Review by tofumatt — CHANGES_REQUESTED** [review]
Just curious if we can lower the amount of time we won't log the same event. It feels like a bit of a brute-force approach, but if it's what needs to be done that's okay.

File: includes/Core/Conversion_Tracking/Conversion_Tracking.php (hunk 2 above)
Comment (zutigrm) [reply includes/Core/Conversion_Tracking/Conversion_Tracking.php]: Thanks, that's a good point, value was in IB, and it slipped my mind that we returned the support for WC, and as you mentioned several events being triggered in shorter time is a realistic possibility. I did several tests with numerous values, and I reached 5ms to be still working. So I updated it for that value.

**Comment by zutigrm** [comment]
Thanks @tofumatt for raising that. I updated the PR

File: includes/Core/Conversion_Tracking/Conversion_Tracking.php (hunk 2 above)
Comment (tofumatt) [reply includes/Core/Conversion_Tracking/Conversion_Tracking.php]: Awesome, thanks! 🙏🏻 

Nice to know we can go that low, given humans can't click that fast I think we're safe 😅

**Review by tofumatt — APPROVED** [review]

**Pull request merged**